docs/source/starthere/fundamentals.rst

```diff
@@ -0,0 +1,240 @@
+	          loss = self.loss(log_probs, labels)
+	          return loss
+
+
+      - .. code-block:: yaml
+	  :caption: Experiment config (YAML)
+	  :linenos:
+	  :emphasize-lines: 4, 7, 10, 13, 16, 20
+
+	  #
+	  # desired configuration of the NeMo model
+	  model:
+	      tokenizer:
+	       ...
+
+	      encoder:
+	       ...
+
+	      decoder:
+	       ...
+
+	      loss:
+	       ...
+
+	      optim:
+	       ...
+
+
+	      train_ds:
+	       ...
+
+	  # desired configuration of the
+	  # PyTorch Lightning trainer object
+	  trainer:
+	      ...
+
+
+Configuring and training NeMo models
+------------------------------------
+
+During initialization of the model, key parameters are read from the config (``cfg``), which gets passed in to the model construtor (left panel above, line 2).
+
+The other object passed into the model's constructor is a PyTorch Lightning ``trainer`` object, which handles the training process. The trainer will take care of the standard training `boilerplate <https://lightning.ai/docs/pytorch/stable/common/trainer.html#under-the-hood>`__. For things that are not standard, PTL will refer to any specific methods that we may have defined in our NeMo model. For example, PTL requires every model to have a specified ``training_step`` method (left panel above, line 15).
+
+The configuration of the trainer is also specified in the config (right panel above, line 20 onwards). This will include parameters such as (number of) ``devices``, ``max_steps``, (numerical) ``precision`` and `more <https://lightning.ai/docs/pytorch/stable/common/trainer.html#trainer-class-api>`__.
+
+
+Example training script
+-----------------------
+
+Putting it all together, here is an example training script for our ``ExampleEncDecModel`` model. We highlight the 3 most important lines, which put together everything we discussed in the previous section.
+
+.. code-block:: python
+	:caption: run_example_training.py
+	:linenos:
+	:emphasize-lines: 10, 11, 12
+
+	import pytorch_lightning as pl
+	from nemo.collections.path_to_model_class import ExampleEncDecModel
+	from nemo.core.config import hydra_runner
+
+	@hydra_runner(
+		config_path="config_file_dir_path",
+		config_name="config_file_name"
+	)
+	def main(cfg):
+		trainer = pl.Trainer(**cfg.trainer)
+		model = ExampleEncDecModel(cfg.model, trainer)
+		trainer.fit(model)
+
+	if __name__ == '__main__':
+		main(cfg)
+
+
+Let's go through the code:
+
+* *Lines 1-3*: import statements (second one is made up for the example).
+* *Lines 5-8*: a decorator on lines 5-8 of ``run_example_training.py`` will look for a config file at ``{config_path}/{config_name}.yaml``, and load its contents into the ``cfg`` object that is passed into the ``main`` function. This functionality is provided by `Hydra <https://hydra.cc/docs/intro/>`__. Instead of a YAML file, we could also have specified the default config as a dataclass, and passed that into the ``@hydra_runner`` decorator.
```

Comment:
remove commas.

* *Lines 5-8*: a decorator on lines 5-8 of ``run_example_training.py`` will look for a config file at ``{config_path}/{config_name}.yaml`` and load its contents into the ``cfg`` object that is passed into the ``main`` function. This functionality is provided by `Hydra <https://hydra.cc/docs/intro/>`__. Instead of a YAML file, we could also have specified the default config as a dataclass and passed that into the ``@hydra_runner`` decorator.